### DISCUSSION TITLE
Localization keys should also accept functions instead of just strings

### DISCUSSION THREAD

**Discussion by gaspardip** [KevinVandy/material-react-table#51] [Ideas]
Title basically, I want to be able to do

```ts
const localization = {
    selectedCountOfRowCountRowsSelected: ({ selectedCount, rowCount }) => `${selectedCount} de ${rowCount} fila(s) seleccionadas`
}
```

This gives users a lot more flexibility about their translations.

A few use cases are

* Pluralizing strings
* Dynamic strings

**Comment by KevinVandy** [comment]
Possibly a good idea, though you can get access to the state by hoisting, or looking at the tableInstanceRef to get those parameters. See [here](https://www.material-react-table.com/docs/guides/table-state-management#access-the-underlying-table-instance-reference)

**Reply by KevinVandy** [reply]
Would you be interested in helping with a PR if going this route? I think I might have to make it so that it accepts a string or a function for each translation with vars

**Reply by gaspardip** [reply]
Yeah, give me a few days, I can create a PR.

**Reply by gaspardip** [reply]
Were you thinking of doing it at the component level or use a context provider and consume it in the respective components?
